eclass/linux-mod-r1.eclass

```diff
@@ -904,6 +935,27 @@ _modules_process_depmod.d() {
 	)
 }
 
+# @FUNCTION: _modules_process_dracut.conf.d
+# @USAGE: <module>...
+# @INTERNAL
+# @DESCRIPTION:
+# Create dracut.conf.d snippet defining if module should be included in the
+# initramfs.
+_modules_process_dracut.conf.d() {
+	(
+		insinto /usr/lib/dracut/dracut.conf.d
+		if [[ ${MODULES_IUSE_INITRAMFS} ]]; then
+			newins - 10-${PN}.conf <<-EOF || die
+				$(usex initramfs {add,omit}_drivers)="${@%.ko}"
+			EOF
+		else
+			newins - 10-${PN}.conf <<-EOF || die
+				omit_drivers="${@%.ko}"
+			EOF
+		fi
+	)
+}
```

Comment:
In this new context I think using a temporary variable would be nicer than repeating the newins or having an overcomplex `$(...)` to inline.

Think I'd personally do:
```bash
	(
		insinto /usr/lib/dracut/dracut.conf.d
		[[ ${MODULES_IUSE_INITRAMFS} ]] && use initramfs && : add || : omit
		newins - 10-${PN}.conf <<<"${_}_drivers=\"${*%.ko}\""
	)
```
Also removes a subshell fwiw, and given how short it is there's not much need for EOF.

On a side-note, there was an unnecessary `|| die` there (given this is not `cat`)

Reply:
Cool, this is a lot shorter, Thanks!